contracts/CTKNToken.sol

```diff
@@ -0,0 +1,357 @@
+        _;
+    }
+
+    /**
+     * As each token is minted it is added to the stockholders array.
+     * @param _to The address that will receive the minted tokens.
+     * @param _amount The amount of tokens to mint.
+     * @return A boolean that indicates if the operation was successful.
+     */
+    function mint(address _to, uint256 _amount)
+        public
+        onlyOwner
+        canMint
+        isVerifiedAddress(_to)
+        returns (bool)
+    {
+        // if the address does not already own stock then
+        // add the address to the stockholders array and record the index.
+        updateStockholders(_to);
+        return super.mint(_to, _amount);
+    }
+
+    /**
+     *  The number of addresses that own tokens.
+     *  @return the number of unique addresses that own tokens.
+     */
+    function holderCount()
+        public
+        onlyOwner
+        view
+        returns (uint)
+    {
+        return stockholders.length;
+    }
+
+    /**
+     *  By counting the number of Token holders using `holderCount`
+     *  you can retrieve the complete list of Token holders, one at a time.
+     *  It MUST throw if `index >= holderCount()`.
+     *  @param index The zero-based index of the holder.
+     *  @return the address of the Token holder with the given index.
+     */
+    function holderAt(uint256 index)
+        public
+        onlyOwner
+        view
+        returns (address)
+    {
+        require(index < stockholders.length);
+        return stockholders[index];
+    }
+
+    /**
+     *  Add a verified address, along with an associated verification hash to the contract.
+     *  Upon successful addition of a verified address the contract must emit
+     *  `VerifiedAddressAdded(addr, hash, msg.sender)`.
+     *  It MUST throw if the supplied address or hash are zero, or if the address has already been supplied.
+     *  @param addr The address of the person represented by the supplied hash.
+     *  @param hash A cryptographic hash of the address holder's verified information.
+     */
+    function addVerified(address addr, bytes32 hash)
+        public
+        onlyOwner
+        isNotCancelled(addr)
+    {
+        require(addr != ZERO_ADDRESS);
+        require(hash != ZERO_BYTES);
+        require(verified[addr] == ZERO_BYTES);
+        verified[addr] = hash;
+        VerifiedAddressAdded(addr, hash, msg.sender);
+    }
+
+    /**
+     *  Remove a verified address, and the associated verification hash. If the address is
+     *  unknown to the contract then this does nothing. If the address is successfully removed this
+     *  function must emit `VerifiedAddressRemoved(addr, msg.sender)`.
+     *  It MUST throw if an attempt is made to remove a verifiedAddress that owns Tokens.
+     *  @param addr The verified address to be removed.
+     */
+    function removeVerified(address addr)
+        public
+        onlyOwner
+    {
+        require(balances[addr] == 0);
```

Comment:
Do we need require(addr != ZERO_ADDRESS); before this line?

Reply:
no because `balances[0x0]` must be `0`